presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java

```diff
@@ -514,11 +519,21 @@ else if (type instanceof DecimalType) {
         }
     }
 
+    private Object getPrimitiveObjectValue(int column, Object fieldData)
+    {
+        try {
+            return ((PrimitiveObjectInspector) fieldInspectors[column]).getPrimitiveJavaObject(fieldData);
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(HIVE_BAD_DATA, String.format("Error parsing field value '%s' for field %s: %s", fieldData, column, e.getMessage()), e);
```

Comment:
`e.getMessage()` may be null, which leads to funny messages `"Error parsing: null"`.
You can resolve this with `firstNonNull(e.getMessage(), e)`.
(usually we do not include the message directly in the top-level exception's message, but i still find it valuable at times.)

Nit: per code-style, we import `format` statically.

Reply:
It's probably fine to exclude the original error message from the re-thrown error, since it's usually obvious (something is not a valid number), and the original error is available in the stack trace.
```java
throw new PrestoException(HIVE_BAD_DATA, format("Error parsing field value '%s' for field %s", fieldData, column), e);
```
But I'm fine with including it you think it is valuable to diagnose errors.